backend/apps/schemes/services/benefit_service.py

```
@@ -141,6 +152,18 @@ def benefit_update(*, benefit_id: str, update_data: dict, user=None):
                     "Another benefit with this name and patient type already exists"
                 )
 
+        # Handle plan field conversion
+        if 'plan' in update_data:
+            if update_data['plan']:
+                from apps.schemes.models import Plan
+                try:
+                    plan = Plan.objects.get(id=update_data['plan'])
+                    update_data['plan'] = plan
+                except Plan.DoesNotExist:
+                    raise ValidationError(f"Plan with id {update_data['plan']} does not exist.")
```

Ruff (0.14.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)